### PR TITLE
LPD-3185 At least 2 chars for searching

### PR DESCRIPTION
**Pull request by michelevigilante**
https://liferay.atlassian.net/browse/LPD-3185

### CODE REVIEW

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:playwright

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:playwright

**Comment by michelevigilante** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Comment by riccardo-alberti** [comment]
ci:forward

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	) => Promise<Locator>;
+	readonly page: Page;
+
+	constructor(page: Page) {
+		this.cleanGlobalSearchBarButton = page.getByLabel('Clear Search');
+
+		this.globalSearchBarCatalogPageButton = page.locator(
+			'.commerce-topbar-button'
+		);
+		this.globalSearchBarCatalogPage = page
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:24]: globalSearchBarInput

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			.locator('#search-bar')
+			.getByPlaceholder('Search');
+
+		this.globalSearchResults = page.locator(
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:28]: globalSearchCommerceItemLink

Comment (alessiorendina) [reply modules/test/playwright/pages/commerce/commerceCatalogPage.ts:28]: Try `getByRole('link', {name: xyz})`

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			`xpath=//div[@class="commerce-suggestions"]//div[@class="commerce-item commerce-item--search"]`
+		);
+
+		this.productDetailProductName = async (productName) => {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:32]: productDetailProductNameHeading

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+			return page.getByRole('heading', {name: productName});
+		};
+
+		this.page = page;
+	}
+
+	async globalSearch(query: string) {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:39]: `search`

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		await this.globalSearchBarCatalogPage.waitFor({state: 'visible'});
+
+		await this.globalSearchBarCatalogPage.fill(query);
+	}
+
+	async findInSearchResult(text: string): Promise<Locator> {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:45]: Move in the Locators:

`readonly globalSearchCommerceItemRow: (text: string) => Promise<Locator>;`

try to check if you can use `getByRole`

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		return this.page
+			.locator(`xpath=//div[@class="commerce-suggestions"]//div/p`)
+			.getByText(text);
+	}
+
+	async openGlobalSearchBarCatalogPage() {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:51]: focusGlobalSearchBarInput

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		await this.page.waitForSelector(
+			'.commerce-topbar-button.js-toggle-search'
+		);
+
+		await this.globalSearchBarCatalogPageButton.click();
+	}
+
+	async waitForSearchResults() {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:59]: Try to move this into Locators

File: modules/test/playwright/tests/commerce/commerceCatalog.spec.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+	commerceCatalogPage,
+	page,
+}) => {
+	const site = await apiHelpers.headlessSite.createSite({
+		name: 'Minium',
+		templateKey: 'minium-initializer',
+		templateType: 'site-initializer',
+	});
+
+	await apiHelpers.commerceMiniumApiHelper.loadCleanUpData(site);
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerceCatalog.spec.ts:34]: Use the utility done here: https://github.com/liferay-commerce/liferay-portal/pull/4994/commits/63db7c5f40383f785c331923be43682541ea2f0a

File: modules/test/playwright/helpers/CommerceMiniumApiHelper.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+export class CommerceMiniumApiHelper {
+	readonly apiHelpers: ApiHelpers | DataApiHelpers;
+	readonly dataApiHelpers: DataApiHelpers;
+
+	constructor(apiHelpers: ApiHelpers | DataApiHelpers) {
+		this.apiHelpers = apiHelpers;
+	}
+
+	async loadCleanUpData(site: Site) {
```
Comment (alessiorendina) [inline modules/test/playwright/helpers/CommerceMiniumApiHelper.ts:16]: Use the utility done here: https://github.com/liferay-commerce/liferay-portal/pull/4994/commits/63db7c5f40383f785c331923be43682541ea2f0a

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@michelevigilante please check my comments, thanks!

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+	readonly productDetailProductNameHeading: (
+		productName: string
+	) => Promise<Locator>;
+	readonly page: Page;
+
+	constructor(page: Page) {
+		this.cleanGlobalSearchBarButton = page.getByRole('button', {
+			name: 'Clear Search',
+		});
+
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:22]: Remove empty line

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+		this.globalSearchBarButton = page
+			.locator('.commerce-topbar-button__icon')
+			.first();
+
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:26]: Remove empty line

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+		this.globalSearchCommerceItemRow = async (text) => {
+			return this.page.getByRole('link', {name: text});
+		};
+		this.globalSearchBarInput = page
+			.locator('#search-bar')
+			.getByPlaceholder('Search');
+		this.productDetailProductNameHeading = async (productName) => {
+			return page.getByRole('heading', {name: productName});
+		};
+		this.page = page;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:36]: Sort alphabetically

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+	readonly globalSearchCommerceItemRow: (text: string) => Promise<Locator>;
+	readonly globalSearchBarButton: Locator;
+	readonly globalSearchBarInput: Locator;
+	readonly productDetailProductNameHeading: (
+		productName: string
+	) => Promise<Locator>;
+	readonly page: Page;
+
+	constructor(page: Page) {
+		this.cleanGlobalSearchBarButton = page.getByRole('button', {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:19]: clearSearchButton

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+		this.globalSearchCommerceItemRow = async (text) => {
+			return this.page.getByRole('link', {name: text});
+		};
+		this.globalSearchBarInput = page
+			.locator('#search-bar')
+			.getByPlaceholder('Search');
+		this.productDetailProductNameHeading = async (productName) => {
+			return page.getByRole('heading', {name: productName});
+		};
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:35]: Move into productDetailsPage

File: modules/test/playwright/pages/commerce/commerceCatalogPage.ts [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+		this.page = page;
+	}
+
+	async search(query: string) {
+		await this.globalSearchBarInput.waitFor({state: 'visible'});
+
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceCatalogPage.ts:41]: Remove empty line

File: modules/test/playwright/tests/commerce/commerceCatalog.spec.ts [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+	});
+
+	apiHelpers.data.push({id: site.id, type: 'site'});
+
+	await commerceLayoutsPage.cleanupSiteInitializerData(apiHelpers, site.name);
+
+	await applicationsMenuPage.goToSite(site.name);
+
+	await commerceCatalogPage.focusGlobalSearchBarInput();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerceCatalog.spec.ts:41]: Remove empty line

File: modules/test/playwright/tests/commerce/commerceCatalog.spec.ts [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+			'Wear Sensors Product designed'
+		)
+	).waitFor({state: 'visible'});
+
+	expect(
+		await commerceCatalogPage.globalSearchCommerceItemRow(
+			'ABS Sensor Product designed'
+		)
+	).toBeVisible();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerceCatalog.spec.ts:63]: Remove empty line

File: modules/test/playwright/tests/commerce/commerceCatalog.spec.ts [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+	expect(
+		await commerceCatalogPage.globalSearchCommerceItemRow(
+			'Wear Sensors Product designed'
+		)
+	).toBeVisible();
+
+	await commerceCatalogPage.cleanGlobalSearchBarButton.click();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerceCatalog.spec.ts:71]: Remove empty line

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@michelevigilante please check my comments, thanks!

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@michelevigilante please check my comments, thanks!

**Review by alessiorendina — APPROVED** [review]

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:forward

**Pull request closed without merging**